core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/SnowflakeCallMapper.scala

```diff
@@ -1,16 +1,111 @@
-        DateAdd(args.head, args(1))
-      case CallFunction("REGEXP_LIKE", args) =>
-        RLike(args.head, args(1))
-      case x: CallFunction => super.convert(x)
+  override def convert(call: ir.Fn): ir.Expression = {
+    withNormalizedName(call) match {
+      case ir.CallFunction("ARRAY_CAT", args) => ir.Concat(args)
+      case ir.CallFunction("ARRAY_CONSTRUCT", args) => ir.CreateArray(args)
+      case ir.CallFunction("BOOLAND_AGG", args) => ir.BoolAnd(args.head)
+      case ir.CallFunction("DATEADD", args) => ir.DateAdd(args.head, args(1))
+      case ir.CallFunction("EDITDISTANCE", args) => ir.Levenshtein(args.head, args(1))
+      case ir.CallFunction("IFNULL", args) => ir.Coalesce(args)
+      case ir.CallFunction("JSON_EXTRACT_PATH_TEXT", args) => getJsonObject(args)
+      case ir.CallFunction("LEN", args) => ir.Length(args.head)
+      case ir.CallFunction("LISTAGG", args) => ir.ArrayJoin(args.head, ir.CollectList(args(1), None), None)
+      case ir.CallFunction("MONTHNAME", args) => ir.DateFormatClass(args.head, ir.Literal("MMM"))
+      case ir.CallFunction("OBJECT_KEYS", args) => ir.JsonObjectKeys(args.head)
+      case ir.CallFunction("POSITION", args) => ir.CallFunction("LOCATE", args)
+      case ir.CallFunction("REGEXP_LIKE", args) => ir.RLike(args.head, args(1))
+      case ir.CallFunction("SPLIT_PART", args) => splitPart(args)
+      case ir.CallFunction("SQUARE", args) => ir.Pow(args.head, ir.Literal(2))
+      case ir.CallFunction("STRTOK_TO_ARRAY", args) => split(args)
+      case ir.CallFunction("TO_DOUBLE", args) => ir.CallFunction("DOUBLE", args)
+      case ir.CallFunction("TO_NUMBER", args) => toNumber(args)
+      case ir.CallFunction("TO_OBJECT", args) => ir.StructsToJson(args.head, args(1))
+      case ir.CallFunction("TRY_TO_NUMBER", args) => tryToNumber(args)
+      case x => super.convert(x)
     }
   }
+
+  private def getJsonObject(args: Seq[ir.Expression]): ir.Expression = {
+    val translatedFmt = args match {
+      case Seq(_, StringLiteral(path)) => ir.Literal("$." + path)
+      case Seq(_, id: ir.Id) => ir.Concat(Seq(ir.Literal("$."), id))
+      case a => throw TranspileException(s"Unsupported arguments to GET_JSON_OBJECT: ${a.mkString("(", ", ", ")")}")
+    }
+    ir.GetJsonObject(args.head, translatedFmt)
+  }
+
+  private def split(args: Seq[ir.Expression]): ir.Expression = {
+
+    val delim = args(1) match {
+      case StringLiteral(d) => ir.Literal(s"[$d]")
+      case e => ir.Concat(Seq(ir.Literal("["), e, ir.Literal("]")))
+    }
+    ir.StringSplit(args.head, delim, None)
+  }
+
+  private def toNumber(args: Seq[ir.Expression]): ir.Expression = {
+    val getArg: Int => Option[ir.Expression] = args.lift
+    if (args.size < 2) {
+      throw TranspileException("not enough arguments to TO_NUMBER")
+    } else if (args.size == 2) {
+      ir.ToNumber(args.head, args(1))
+    } else {
+      val fmt = getArg(1).collect { case f @ StringLiteral(_) =>
+        f
```

Comment:
what would happen if the argument is not the string literal? panic? computer reboot?

Reply:
`fmt` will be `None`

Reply:
make a default value for the format.